src/adapters/voice-db.ts

```diff
@@ -0,0 +1,35 @@
+import SQL from 'sql-template-strings'
+import { randomUUID } from 'node:crypto'
+import { AppComponents, IVoiceDatabaseComponent } from '../types'
+import { normalizeAddress } from '../utils/address'
+
+export async function createVoiceDBComponent(
+  components: Pick<AppComponents, 'pg' | 'config'>
+): Promise<IVoiceDatabaseComponent> {
+  const { pg, config } = components
+  // Private voice chat expiration time in milliseconds
+  const PRIVATE_VOICE_CHAT_EXPIRATION_TIME = await config.requireNumber('PRIVATE_VOICE_CHAT_EXPIRATION_TIME')
+
+  return {
+    async areUsersBeingCalledOrCallingSomeone(userAddresses: string[]): Promise<boolean> {
+      const normalizedUserAddresses = userAddresses.map(normalizeAddress)
+
+      const query = SQL`
+        SELECT EXISTS (
+          SELECT 1 FROM private_voice_chats WHERE caller_address IN (${normalizedUserAddresses}) OR callee_address IN (${normalizedUserAddresses})
+        )
+      `
+      const results = await pg.query<{ exists: boolean }>(query)
+      return results.rows[0].exists
```

Comment:
I added a method `exists` in a decorated `pg` component that you could use here if you want.